interfaces/sourcegen/sourcegen/_orchestrate.py

```diff
@@ -4,40 +4,47 @@
 import importlib
 import inspect
 from pathlib import Path
+import logging
+import sys
 from typing import List, Dict
-import ruamel.yaml
 
 from ._HeaderFileParser import HeaderFileParser
 from ._SourceGenerator import SourceGenerator
+from ._helpers import read_config
 
 
-_clib_path = Path(__file__).parent.joinpath("../../../include/cantera/clib").resolve()
-_clib_defs_path = _clib_path.joinpath("clib_defs.h")
+_logger = logging.getLogger()
 
-def generate_source(lang: str, out_dir: str):
-    print("Generating source files...")
+class CustomFormatter(logging.Formatter):
+    """Minimalistic logging output"""
+
+    def format(self, record):
+        formatter = logging.Formatter("[%(levelname)s] %(message)s")
+        return formatter.format(record)
 
-    module = importlib.import_module(__package__ + "." + lang)
-    config_path = Path(module.__file__).parent.joinpath("config.yaml")
 
-    config = {}
-    if config_path.exists():
-        with config_path.open() as config_file:
-            reader = ruamel.yaml.YAML(typ="safe")
-            config = reader.load(config_file)
+def generate_source(lang: str, out_dir: str=""):
+    """Main entry point of sourcegen."""
+    loghandler = logging.StreamHandler(sys.stdout)
+    loghandler.setFormatter(CustomFormatter())
+    _logger.handlers.clear()
+    _logger.addHandler(loghandler)
+    _logger.setLevel(logging.DEBUG)
+    _logger.info(f"Generating {lang!r} source files...")
 
-    ignore_files: List[str] = config.get("ignore_files", [])
-    ignore_funcs: Dict[str, List[str]] = config.get("ignore_funcs", {})
+    module = importlib.import_module(__package__ + "." + lang)
+    root = Path(module.__file__).parent
+    config = read_config(root.joinpath("config.yaml"))
+    templates = read_config(root.joinpath("templates.yaml"))
```

Comment:
You could consider using the overloaded `/` here:
```suggestion
    config = read_config(root / "config.yaml")
    templates = read_config(root / "templates.yaml")
```